### PR TITLE
require PR reviews and linear history

### PR DESCRIPTION
**Pull request by pjfanning**
Based on https://cwiki.apache.org/confluence/display/INFRA/git+-+.asf.yaml+features

The changes may need tweaking - I won't know if the changes will be applied as expected - so I'll fix any issues where expected updates don't actually get applied

https://github.com/apache/incubator-pekko/issues/51

* requires only 1 PR review - https://lists.apache.org/thread/0hohnn9v0xhxn0qblb196p7mcq7jzz00 and https://cwiki.apache.org/confluence/display/PEKKO/Processes requires 1 review for trivial PRs
* tries to enforce that Scalafmt workflow passes - we can add more later
* linear git history

### CODE REVIEW

File: .asf.yaml
Comment (He-Pin) [inline .asf.yaml:42]: at least 2?

Comment (pjfanning) [reply .asf.yaml:42]: only 1 review needed for trivial changes according to https://cwiki.apache.org/confluence/display/PEKKO/Processes - there is no way to work out automatically what is trivial

**Review by mdedetrich — CHANGES_REQUESTED** [review]
Is it possible to add the header check in conjunction with scalafmt?

File: .asf.yaml
Comment (mdedetrich) [reply .asf.yaml:42]: So this is an interesting point. Historically akka always required 2 approvals, even for trivial changes. I will say however in terms of practicality especially right now with a lot of mechanical and trivial changes needed to her the project going 1 should be fine.

We can revisit this process later, I personally think that requiring 2 reviewers once the project is setup has merit especially since at that point in time we will also start seeing non trivial functional changes.

Comment (He-Pin) [reply .asf.yaml:42]: Who is in the position to merge pull requests?

Comment (pjfanning) [reply .asf.yaml:42]: anyone with git commit rights - all Pekko committers can do this - you might need to link your Github and ASF accounts if you haven't done this

**Comment by pjfanning** [comment]
> Is it possible to add the header check in conjunction with scalafmt?

Which of these workflows is the 'header check'? Do we need to add a 'header check' workflow first?

File: .asf.yaml
Comment (He-Pin) [reply .asf.yaml:42]: So the origin author and who proved the PR can merge it too?

**Comment by mdedetrich** [comment]
> Which of these workflows is the 'header check'? Do we need to add a 'header check' workflow first?

its already there, its called  `Pull Requests / Check / Headers`, 

image

File: .asf.yaml
Comment (pjfanning) [reply .asf.yaml:43]: I don't know how to implement:

```
Any review that has open requests or discussion blocks the merge. (only merge when consensus is reached).
```

If there is a way, it can be implemented.

Comment (mdedetrich) [reply .asf.yaml:43]: > Any review that has open requests or discussion blocks the merge. (only merge when consensus is reached).

To me I think this is too strict, I have been in cases where pull requests stuck open in perpetuity because someone forgets to click the "resolve" button. Either that or committers will just blindly hit resolve which at that point defeats the purpose of it.

Should we discuss in the mailing list? I think for now this is fine, we can always add stricter checks later if there is a consensus on it.

Comment (mdedetrich) [reply .asf.yaml:43]: I would say lets just just merge the PR as it is now and we can also make a future one to implement these stricter checks.

File: .asf.yaml
Comment (Claudenw) [inline .asf.yaml:31]: Please add:

```## Licensed under the terms of http://www.apache.org/licenses/LICENSE-2.0```

To the top of the file

**Review by Claudenw — CHANGES_REQUESTED** [review]

File: .asf.yaml
Comment (Claudenw) [reply .asf.yaml:43]: @pjfanning I think that

```      required_conversation_resolution: true```

will require the conversation resolution.

Comment (pjfanning) [reply .asf.yaml:43]: I don't see `required_conversation_resolution` in https://cwiki.apache.org/confluence/display/INFRA/git+-+.asf.yaml+features

File: .asf.yaml
Comment (pjfanning) [reply .asf.yaml:46]: done

**Comment by pjfanning** [comment]
@Claudenw Is it possible for you to re-review this?

File: .asf.yaml
Comment (mdedetrich) [inline .asf.yaml:1]: @pjfanning I think you can remove this header change because its going to be done properly in https://github.com/apache/incubator-pekko/pull/50

File: .asf.yaml
Comment (mdedetrich) [inline .asf.yaml:52]: As per https://github.com/apache/incubator-pekko-sbt-paradox/pull/7, this needs to be changed to `- Code is formatted` rather than `- Scalafmt`

File: .asf.yaml
Comment (mdedetrich) [inline .asf.yaml:53]: I would recommend checking that this works in a separately. As per https://github.com/apache/incubator-pekko-sbt-paradox/pull/5#issuecomment-1414453790 otherwise we can freeze the main branch and we have to make an INFRA ticket to unfreeze it. The way I did this I pushed a new branch (ergo `test`), created a PR against `main` that only enables the `contexts` check against `test` and then make another PR against test to figure out if the check is satisfied.

**Review by mdedetrich — CHANGES_REQUESTED** [review]
So I have changed my review back to request changes because there if the PR is merged in its current state it would freeze the main branch (check review comments).

**Comment by pjfanning** [comment]
I'll close this as it is stale

**Pull request closed without merging**
